### PR TITLE
Cherry-pick:  Fixed the recursive FLComponents creation

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

- Fixed the recursive FLComponent creation missing the event handling issue.
- Fixed the client job process missing the proper FLcomponent handlers issue.
- Server Runner set the workflow ID in the FLContext before the initialize_run(fl_ctx).
 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/server/server_json_config.py [hunk 1]
```diff
@@ -138,8 +138,8 @@ def _get_all_workflows_ids(self):
             ids.append(t.id)
         return ids
 
-    def authorize_and_build_component(self, config_dict, config_ctx, node):
-        t = super().authorize_and_build_component(config_dict, config_ctx, node)
+    def build_component(self, config_dict):
+        t = super().build_component(config_dict)
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_json_config.py:142]: should we use `authorize_and_build_component` or `build_component`?

File: nvflare/private/fed/client/client_json_config.py [hunk 2]
```diff
@@ -102,8 +102,8 @@ def process_config_element(self, config_ctx: ConfigContext, node: Node):
             self.current_exe.executor = self.authorize_and_build_component(element, config_ctx, node)
             return
 
-    def authorize_and_build_component(self, config_dict, config_ctx, node):
-        t = super().authorize_and_build_component(config_dict, config_ctx, node)
+    def build_component(self, config_dict):
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/client/client_json_config.py:105]: should we use `authorize_and_build_component` or `build_component`?

Comment (yhwen) [reply nvflare/private/fed/client/client_json_config.py:105]: at the entry point, use authorize_and_build_component(). build_component() is used by the recursive calls.

File: nvflare/private/fed/server/server_json_config.py (hunk 1 above)
Comment (yhwen) [reply nvflare/private/fed/server/server_json_config.py:142]: use build_component() here.

**Review by yanchengnv — APPROVED** [review]
LGTM

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**